### PR TITLE
Make Substrate compile with latest nightly

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Review by dvdplm — APPROVED** [review]
LGTM

(If you overrode your nightly rust, don't forget to `rustup default nightly` after installing the latest.)

**Review by cheme — APPROVED** [review]
LGTM

**Comment by pepyakin** [comment]
Care to bump the nix config?

File: primitives/arithmetic/src/fixed_point.rs [hunk 1]
```diff
@@ -536,10 +536,10 @@ macro_rules! implement_fixed {
 			}
 		}
 
-		impl<P: PerThing> From<P> for $name {
+		impl<P: PerThing> From<P> for $name where P::Inner: FixedPointOperand {
 			fn from(p: P) -> Self {
-				let accuracy = P::ACCURACY.saturated_into();
-				let value = p.deconstruct().saturated_into();
+				let accuracy = P::ACCURACY;
```
Comment (andresilva) [inline primitives/arithmetic/src/fixed_point.rs:541]: Does this change anything? What was the type this was saturating into before? Probably dumb question but I don't know this code.

File: client/service/src/client/client.rs [hunk 2]
```diff
@@ -1141,7 +1141,7 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		let mut ancestor = load_header(ancestor_hash)?;
 		let mut uncles = Vec::new();
 
-		for _generation in 0..max_generation.saturated_into() {
+		for _generation in 0u32..UniqueSaturatedInto::<u32>::unique_saturated_into(max_generation) {
```
[the remark this answers is not in the document]
Comment (bkchr) [reply client/service/src/client/client.rs:1144]: This is stable that requires this :shrug:

File: primitives/arithmetic/src/fixed_point.rs (hunk 1 above)
Comment (bkchr) [reply primitives/arithmetic/src/fixed_point.rs:541]: Actually I don't understand how this every compiled. The compiler was just wildly guessing what we want here.

The crucial change here is the where clause that was added above. This makes the difference here ;)

Comment (andresilva) [reply primitives/arithmetic/src/fixed_point.rs:541]: Not sure what was going on here either, the compiler needed to find a way to go from `PerThing` to `FixedPointOperand` 🤷

**Review by andresilva — APPROVED** [review]

**Pull request merged**